### PR TITLE
CMake build structure and style and pkg-config fixes

### PR DESCRIPTION
**Pull request by eaaltonen**
Re-opened after #350 was closed.

Resolve some inconsistencies originating from commit https://github.com/OP-TEE/optee_client/commit/e0a12e2322ae02e957910f29948bd19455bf9cfa "cmake: Initial CMake support".

I think these would be good to fix in order to:

* improve the consistency
* prepare for adding a CMake config-file package to the install
* install the `teec` pkgconfig file

### CODE REVIEW

**Review by jforissier — COMMENTED** [review]
For "cmake: pkgconfig file: install the file":

`Acked-by: Jerome Forissier <jerome.forissier@linaro.org>`

**Comment by eaaltonen** [comment]
Squeashed fixup.

>  For "cmake: pkgconfig file: install the file":
> 
> Acked-by: Jerome Forissier <jerome.forissier@linaro.org>

added.

> I believe include/linux/tee.h and include/teec_benchmark.h should be moved under src. They are not part of the libteec API.

Moved, thanks.

**Review by jforissier — COMMENTED** [review]
Thanks. For "Makefile, cmake: move teec related headers":

`Acked-by: Jerome Forissier <jerome.forissier@linaro.org>`

**Review by etienne-lms — COMMENTED** [review]
`Acked-by: Etienne Carriere <etienne.carriere@foss.st.com>` for the whole series.

**Comment by eaaltonen** [comment]
`Acked-by`, thanks.

File: CMakeLists.txt [hunk 1]
```diff
@@ -40,7 +40,7 @@ endif(CCACHE_FOUND)
 
 add_subdirectory(libteec)
 add_subdirectory(tee-supplicant)
-add_subdirectory(public)
+
```
Comment (eaaltonen) [inline CMakeLists.txt:43]: I'll remove this still.

**Pull request merged**